src/query/block/lazy_test.go

```diff
@@ -418,3 +419,224 @@ func TestUnconsolidatedSeriesIter(t *testing.T) {
 
 	assert.Equal(t, expected, actual.Datapoints())
 }
+
+// negative value offset tests
+
+func TestStepIterWithNegativeValueOffset(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+	b := NewMockBlock(ctrl)
+	offset := time.Duration(0)
+	off := NewLazyBlock(b, testLazyOpts(offset, -1.0))
+	msg := "err"
+	e := errors.New(msg)
+	now := time.Now()
+
+	iter := NewMockStepIter(ctrl)
+	b.EXPECT().StepIter().Return(iter, nil)
+	it, err := off.StepIter()
+	require.NoError(t, err)
+
+	// ensure functions are marshalled to the block's underlying step iterator.
+	iter.EXPECT().Close()
+	it.Close()
+
+	iter.EXPECT().Err().Return(e)
+	assert.EqualError(t, it.Err(), msg)
+
+	iter.EXPECT().StepCount().Return(12)
+	assert.Equal(t, 12, it.StepCount())
+
+	seriesMetas := []SeriesMeta{}
+	iter.EXPECT().SeriesMeta().Return(seriesMetas)
+	assert.Equal(t, seriesMetas, it.SeriesMeta())
+
+	iter.EXPECT().Next().Return(true)
+	assert.True(t, it.Next())
+
+	vals := []float64{1, 2, 3}
+	step := NewMockStep(ctrl)
+	step.EXPECT().Values().Return(vals)
+	step.EXPECT().Time().Return(now)
+
+	expectedVals := []float64{-1, -2, -3}
+	iter.EXPECT().Current().Return(step)
+	actual := it.Current()
+	assert.Equal(t, expectedVals, actual.Values())
+	assert.Equal(t, now, actual.Time())
+}
+
+func TestSeriesIterWithNegativeValueOffset(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+	b := NewMockBlock(ctrl)
+	offset := time.Duration(0)
+	off := NewLazyBlock(b, testLazyOpts(offset, -1.0))
```

Comment:
nit: can do `testLazyOpts(0, -1.0)` and it'll typecast it properly

Reply:
yeah just figured I'd be more explicit with it being a duration.